Add initial Brazil NFSe addon

cavalle opened the pull request:
## Describe your changes

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] I have added thorough tests with at **least** 90% code coverage.
- [x] I've modified or created example GOBL documents the show my changes in use, if appropriate.
- [x] When adding or modifying a tax regime or addon, I've added links to the source of the changes either structured or in the comments.
- [x] I've run `go generate .` to ensure that Schemas and Regime data are up to date.
- [x] All linter warnings have been reviewed and fixed.
- [x] I've been obsessive with pointer nil checks to avoid panics.
- [x] The CHANGELOG.md has been updated with an overview of my changes.
- [x] Requested a review from @samlown.

samlown commented on addons/br/nfse/extensions.go:
```diff
@@ -0,0 +1,32 @@
+
+				* https://www.planalto.gov.br/ccivil_03/leis/lcp/lcp116.htm
+			`),
+		},
```
Is is possible to add a `Pattern` flag perhaps to validate the codes?

cavalle replied:
I chose not to validate anything since every municipality can define their own list of codes without restrictions on the format. The list in LEI116 is commonly used, apparently, but there's no guarantee those will be the codes in the municipality.

samlown commented on addons/br/nfse/line.go:
```diff
@@ -0,0 +1,18 @@
+	}
+
+	return validation.Validate(line,
+		bill.RequireLineTaxCategory(br.TaxCategoryISS),
```
Does this also need to be repeated for Discount and Charge and lines?

cavalle replied:
Actually, I don't think it makes sense to support either Charges or Discounts: it doesn't seem possible to map them to an NFSe. So, I've added validations to ensure those aren't present in [177eb69](https://github.com/invopop/gobl/pull/402/commits/177eb697a8a4bf0fecd5edf6bc1cab717d7fa216)

samlown commented on addons/br/nfse/item_test.go:
```diff
@@ -0,0 +1,62 @@
+				},
+			},
+			err: "ext: (br-nfse-service: required.).",
+		},
```
You could test the nil case here like this:
```suggestion
		},
		{
		  name: "nil",
		  item: nil,
		},
```

samlown reviewed (approved):
Nice. Some observations. Do the Charges and Discounts also need to be covered?

cavalle commented:
@samlown Thanks for your feedback! I think I addressed all your points and made an additional change (derived from your feedback as well) that let me get to 98% coverage.
